xchange-deribit/src/main/java/org/knowm/xchange/deribit/v2/service/DeribitMarketDataServiceRaw.java

```diff
@@ -0,0 +1,121 @@
+package org.knowm.xchange.deribit.v2.service;
+
+import java.io.IOException;
+import java.util.List;
+import org.knowm.xchange.deribit.v2.DeribitErrorAdapter;
+import org.knowm.xchange.deribit.v2.DeribitExchange;
+import org.knowm.xchange.deribit.v2.dto.marketdata.*;
+import org.knowm.xchange.deribit.v2.dto.marketdata.response.*;
+import si.mazi.rescu.HttpStatusIOException;
+
+/**
+ * Implementation of the market data service for Bitmex
+ *
+ * <ul>
+ *   <li>Provides access to various market data values
+ * </ul>
+ */
+@SuppressWarnings({"WeakerAccess", "JavaDoc"})
+public class DeribitMarketDataServiceRaw extends DeribitBaseExchange {
+
+  /**
+   * Constructor
+   *
+   * @param exchange
+   */
+  public DeribitMarketDataServiceRaw(DeribitExchange exchange) {
+
+    super(exchange);
+  }
+
+  public List<DeribitInstrument> getDeribitActiveInstruments(String currency) throws IOException {
+
+    DeribitInstrumentsResponse response = null;
+
+    try {
+      response = deribit.getInstruments(currency);
+    } catch (HttpStatusIOException ex) {
```

Comment:
when you implement my suggestions with the DeribitException then it will get thrown here instead

methods in *ServiceRaw classes should throw out DeribitException 

they should be caught and adapted in *Service methods

Reply:
Ok, but it will cause that some *ServiceRaw methods will miss adaptation (e,g. getDeribitActiveInstruments method), right?

Reply:
yes, that's the point, raw methods should use exchanges DTO (DeribitException is also somewhat a DTO for error situations), adaption should happen in methods that implement generic interfaces

Reply:
Ok, now i get it. Thanks.